src/main/java/com/dao/UserDAO.java

```diff
@@ -0,0 +1,205 @@
+package com.dao;
+
+import com.model.User;
+
+import javax.sql.DataSource;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserDAO {
+
+    private static UserDAO instance;
+    private DataSource dataSource;
+
+    private static final String IS_EXIST_QUERY = "SELECT * FROM users WHERE user_name = ?";
+    private static final String VALIDATION_QUERY = "SELECT * FROM users WHERE user_name = ? AND password_hash = ?";
+    private static final String GET_BY_NAME_QUERY = "SELECT * FROM users WHERE user_name = ?";
+    private static final String ADD_QUERY = "INSERT INTO users VALUES (NULL, ?, ?, ?)";
+    private static final String REMOVE_QUERY = "DELETE FROM users WHERE user_name = ?";
+    private static final String UPDATE_QUERY = "UPDATE users SET is_admin = ?, password_hash = ? WHERE user_name = ?";
+    private static final String GET_ALL_QUERY = "SELECT * FROM users";
+
+    private UserDAO() {
+    }
+
+    public static UserDAO getInstance() {
+
+        if (instance == null) {
+
+            instance = new UserDAO();
+        }
+
+        return instance;
+    }
+
+    public void setDataSource(DataSource dataSource) {
+
+        this.dataSource = dataSource;
+    }
+
+    public boolean isExist(String userName) {
+
+        boolean isPresent = false;
+
+        try (Connection connection = dataSource.getConnection()) {
+
+            PreparedStatement sql = connection.prepareStatement(IS_EXIST_QUERY);
+            sql.setString(1, userName);
+            ResultSet rs = sql.executeQuery();
+
+            isPresent = rs.next();
+
+        } catch (SQLException e) {
+
+            e.printStackTrace();
+        }
+
+        return isPresent;
+    }
+
+    public boolean validate(String userName, String password) {
+
+        boolean isPresent = false;
+
+        try (Connection connection = dataSource.getConnection()) {
+
+            PreparedStatement sql = connection.prepareStatement(VALIDATION_QUERY);
+            sql.setString(1, userName);
+            sql.setString(2, password);
+            ResultSet rs = sql.executeQuery();
+
+            isPresent = rs.next();
+
+        } catch (SQLException e) {
+
+            e.printStackTrace();
+        }
+
+        return isPresent;
+    }
+
+    public User getByName(String userName) {
+
+        User user = null;
```

Comment:
It's bad practice to return nulls. Consider usage of Optional class that was introduced in java 8